gcp/redis_location_list.go

```diff
@@ -0,0 +1,56 @@
+package gcp
+
+import (
+	"context"
+
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+	"google.golang.org/api/iterator"
+	"google.golang.org/genproto/googleapis/cloud/location"
+)
+
+const matrixKeyRedisLocation = "redis-location"
+
+func BuildRedisLocationList(ctx context.Context, d *plugin.QueryData) []map[string]interface{} {
+
+	// have we already created and cached the locations?
+	locationCacheKey := "RedisLocation"
+	if cachedData, ok := d.ConnectionManager.Cache.Get(locationCacheKey); ok {
+		plugin.Logger(ctx).Trace("listlocationDetails:", cachedData.([]map[string]interface{}))
+		return cachedData.([]map[string]interface{})
+	}
+
+	// Create Service Connection
+	service, err := RedisService(ctx, d)
+	if err != nil {
+		return nil
+	}
+
+	projectData, err := activeProject(ctx, d)
+	if err != nil {
+		return nil
+	}
+	project := projectData.Project
```

Comment:
@madhushreeray30, should we not add an empty check for project value? The function `activeProject()` may return an empty string as a project value if the project is not specified in `gcp.spc` file.